src/engine/Match_extract_mode.mli

```diff
@@ -20,4 +23,7 @@ val extract_nested_lang :
   timeout_threshold:int ->
   Rule.extract_rule list ->
   Xtarget.t ->
-  (Input_to_core_t.target * match_result_location_adjuster) list
+  (Input_to_core_t.target
+  * match_result_location_adjuster
+  * assoc_extracted_target_to_original)
+  list
```

Comment:
Here, I'd return a record and get rid of the type aliases.

Reply:
If the remaining tuples such `string * Fpath.t` are still not obvious, I'd turn them into records as well.